### ISSUE TITLE
Modified options properties are not reflected in the chart

### ISSUE THREAD

**Issue by bjornalm** [carbon-design-system/carbon-charts#817]
#### I am submitting a...

- [ ] Feature request
- [ ] Design defect
- [x] Source code defect
- [ ] Demo/documentation defect
- [ ] Other

#### charts version:
0.38.2 

### Issue description
There are option properties (e.g. `options.legend.position` and `options.data.groupMapsTo`) that will not trigger a re-render. These are the ones I have found so far while working on https://github.com/carbon-design-system/carbon-addons-iot-react/issues/1181, but there might be more.

#### Steps to produce the issue
1. Check out this sandbox
https://codesandbox.io/s/elastic-cloud-q1zi6
2. The timer will change the title of the chart and the alignment of the legend but not the position of the legend. 

#### Current behavior
Changes to the props are not reflected

#### Expected behavior
Changes to the props are reflected

**Comment by theiliad** [comment]
Yes, legend position isn't something we'd anticipated changing in future renders. Currently there isn't any logic in place that'd update that

**Comment by bjornalm** [comment]
> Yes, legend position isn't something we'd anticipated changing in future renders. Currently there isn't any logic in place that'd update that

Changing the legend is perhaps more interesting for demo purposes but the `options.data.groupMapsTo` seems likely to be in need of change after initial rendering, e.g. if the actual data changes.

**Comment by nstuyvesant** [comment]
Confirmed to still be an issue in latest. Here's a [StackBlitz repro](https://stackblitz.com/edit/react-tzm42t-xgmfgs?file=src%2Foptions.js,src%2Findex.js,src%2Fdata.js).
